docs_src/microservices/device/V3Migration.md

```diff
@@ -8,29 +8,145 @@ See [Top Level V3 Migration Guide](../../../V3TopLevelMigration) for details app
 
-TBD
+1. Change device definition file to YAML format.
+2. Remove `LastConnected` and `LastReported` configs.
+3. ProtocolProperties now supports typed values.
+
+    !!! example "ProtocolProperty with typed values"
+    
+        ```yaml
+        protocols:
+          other:
+            Address: simple01
+            Port: 300
+        ```
 
 ### Device Profile Files
 
-TBD
+N/A
 
 ### Provision Watcher files
 
-TBD
+1. ProvisionWatcher can now be added during device service startup by loading the definition files from the `ProvisionWatchersDir` configuration.
+
+    !!! example "Example Configuration"
+    
+        ```yaml
+        Device:
+            ProvisionWatchersDir: ./res/provisionwatchers
+        ```
+
+2. ProvisionWatcher definition file is in YAML format.
+
+    !!! example "Pre-defined ProvisionWatcher"
+
+        ```yaml
+        name: Simple-Provision-Watcher
+        serviceName: device-simple
+        labels:
+          - simple
+        identifiers:
+          Address: simple[0-9]+
+          Port: 3[0-9]{2}
+        blockingIdentifiers:
+          Port:
+            - 397
+            - 398
+            - 399
+        adminState: UNLOCKED
+        discoveredDevice:
+          profileName: Simple-Device
+          adminState: UNLOCKED
+          autoEvents:
+            - interval: 15s
+              sourceName: SwitchButton
+          properties:
+            testPropertyA: weather
+            testPropertyB: meter
+        ```
+
+        Note that in EdgeX 3.0, `ServiceName` is moved out of `DiscoveredDevice` to top level.
 
 ### Other
 
-TBD
+1. The `-c/--confdir` command line option is replaced by `-cd/--configDir`.
+2. The `-f/--file` command line option is replaced by `-cf/--configFile`.
 
 ## Custom Device Services
 
 This section is specific to changes made that impact existing **custom device services**.
 
 See [Top Level V3 Migration Guide](../../../V3TopLevelMigration) for details applicable to all EdgeX services and [All Device Services](#all-device-services) section above for details applicable to all EdgeX device services.  
 
-### Go Device Services
+### Dependencies
+You first need to update the `go.mod` file to specify `go 1.20` and the V3 versions of the Device SDK and any EdgeX go-mods directly used by your service. Note the extra `/v3` for the modules.
+
+!!! example "Example go.mod for V3"
+
+    ```go
+    module <your service>
+    
+    go 1.20
+    
+    require (
+    	github.com/edgexfoundry/device-sdk-go/v3 v3.0.0
+    	github.com/edgexfoundry/go-mod-core-contracts/v3 v3.0.0
+        ...
+    )
+    ```
 
-TBD
+Once that is complete then the import statements for these dependencies must be updated to include the `/v3` in the path.
+
+!!! example "Example import statements for V3"
+
+    ```go
+    import (
+    	...
+        
+    	"github.com/edgexfoundry/device-sdk-go/v3/pkg/models"
+    	"github.com/edgexfoundry/go-mod-core-contracts/v3/common"
+    )
+    ```
+
+### Go Device Services
```

Comment:
add an item to mention about Device Service SDK becomes interface
https://github.com/edgexfoundry/device-sdk-go/pull/1194

Reply:
item added